File: Week 1 C/Problem set 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int owed;
    do
    {
        owed = get_int("Change owed:");
    }
    while (owed < 0);

    int o_quartes = owed / 25;
    owed = owed - (o_quartes * 25);
    int o_dimes = owed / 10;
    owed = owed - (o_dimes * 10);
    int o_nickles = owed / 5;
    owed = owed - (o_nickles * 5);
    int o_pennies = owed / 1;
    owed = owed - (o_pennies * 1);

    printf("%d\n", o_quartes + o_dimes + o_nickles + o_pennies);
}

File: Week 6 Python/Problem set 6/sentimental-cash/cash.py
from cs50 import get_float

while (True):
    owed = get_float("Change: ")
    if owed >= 0:
        break
owed_c = round(owed * 100)

# o_quartes = owed_c // 25
# owed_c = owed_c - (o_quartes * 25)
# o_dimes = owed_c // 10
# owed_c = owed_c - (o_dimes * 10)
# o_nickles = owed_c // 5
# owed_c = owed_c - (o_nickles * 5)
# o_pennies = owed_c // 1
# owed_c = owed_c - (o_pennies * 1)
#
# print(o_quartes + o_dimes + o_nickles + o_pennies)

coins = 0
for coin in [25, 10, 5, 1]:
    coins += owed_c // coin
    owed_c %= coin

print(coins)
